harpoon/suite.go

```diff
@@ -310,18 +323,46 @@ func (s *Suite) RunM(m *testing.M) {
 }
 
 func (s *Suite) RunT(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	var testLog bytes.Buffer
 	if s.output != "" {
 		s.suite.Options.Output = &testLog
 		s.suite.Options.NoColors = true
 	}
 
+	var errMessage internaltesting.TerminationError
+	var wg sync.WaitGroup
```

Comment:
If I'm reading this correctly, wouldn't a single channel be more clear here?

```go
doneCh := make(chan struct{})
var errMsg internaltesting.TerminationError
if s.exitOnCleanUpFailures {
    go func() {
        defer close(doneCh)

        // ... Select here
    }()
} else {
 close(doneCh)
}

cancel()
<-doneCh
```

A wait group has me thinking "Where's the loop? I'm clearly missing the loop" (Am I just missing the loop?)

Reply:
Nope, no loop -- I can do a channel instead 🙂